subprojects/tooling-api/src/main/java/org/gradle/tooling/TestPatternSpec.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling;
+
+import org.gradle.api.Incubating;
+
+import java.util.Collection;
+
+/**
+ * Provides fine-grained test selection within a test task.
+ *
+ * @since 7.5
+ */
+@Incubating
+public interface TestPatternSpec {
```

Comment:
Do the package, classes and methods also accept patterns implicitly?

Reply:
That is the case. The configuration in `TestPatternSpec` configures the `TestFilter`s on the provider side, which are converted to test patterns.

Reply:
I've documented this in the javadoc.